frontend/src/components/Header.tsx

```diff
@@ -0,0 +1,57 @@
+import { faMoon, faSun } from '@fortawesome/free-regular-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { headerLinks } from '../utils/constants';
+import { NavLink } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+export default function Header() {
+    const [dark, setDark] = useState(() => {
+        return localStorage.getItem("theme") === "dark";
```

Comment:
I have implemented a re-usable `ModeToggle.tsx` component in my PR. I guess it can be used in here

Reply:
@harsh3dev Thank you for letting me know.  I'm not sure however if we actually need a reusable component for this, since it's only appearing in one place in the Header component at all times. 🤔